provider/cluster_rosa_classic_resource.go

```diff
@@ -645,20 +670,40 @@ func createClassicClusterObject(ctx context.Context,
 	return object, err
 }
 
-func (r *ClusterRosaClassicResource) validateAccountRoles(ctx context.Context, state *ClusterRosaClassicState) error {
-	r.logger.Debug(ctx, "Validating if cluster version is compatible to account roles' version")
-	region := state.CloudRegion.Value
-	version := ""
+// getAndValidateVersion ensures that the cluster version is available in the
+// channel group
+func (r *ClusterRosaClassicResource) getAndValidateVersion(ctx context.Context, state *ClusterRosaClassicState) (string, error) {
```

Comment:
The function name might be confusing
Maybe change the function name to something like validateClusterVersionCompatibleWithChannelGroup... what do you think?

Reply:
This function could also help us if the cluster version and channel group were set. (in the create)

Reply:
renamed